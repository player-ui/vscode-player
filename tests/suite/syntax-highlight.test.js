const { expect } = require("expect");
const path = require("path");
const fs = require("fs");
const { commands, Uri } = require("vscode");

const syntaxHighlightSampleDir = path.resolve(
  __dirname,
  "./syntax-highlight-samples"
);
const files = fs.readdirSync(syntaxHighlightSampleDir);

async function getColorTokens(filePath) {
  const data = await commands.executeCommand(
    "_workbench.captureSyntaxTokens",
    Uri.file(filePath)
  );
  return data;
}

suite("Syntax Highlight", () => {
  files.forEach((file) => {
    const filePath = path.join(syntaxHighlightSampleDir, file);
    // const fileContent = fs.readFileSync(filePath, 'utf8');
    test(`Syntax Highlight: ${file}`, async function () {
      expect(await getColorTokens(filePath)).toMatchSnapshot(this);
    });
  });
});
