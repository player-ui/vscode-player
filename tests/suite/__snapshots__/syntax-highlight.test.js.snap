// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Syntax Highlight Syntax Highlight: basic-binding.js 1`] = `
Array [
  Object {
    "c": "const",
    "r": Object {
      "dark_plus": "storage.type: #569CD6",
      "dark_vs": "storage.type: #569CD6",
      "hc_black": "storage.type: #569CD6",
      "hc_light": "storage.type: #0F4A85",
      "light_plus": "storage.type: #0000FF",
      "light_vs": "storage.type: #0000FF",
    },
    "t": "source.js meta.var.expr.js storage.type.js",
  },
  Object {
    "c": " ",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js",
  },
  Object {
    "c": "foo",
    "r": Object {
      "dark_plus": "variable.other.constant: #4FC1FF",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "variable: #9CDCFE",
      "hc_light": "variable.other.constant: #02715D",
      "light_plus": "variable.other.constant: #0070C1",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js meta.var-single-variable.expr.js meta.definition.variable.js variable.other.constant.js",
  },
  Object {
    "c": " ",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js meta.var-single-variable.expr.js",
  },
  Object {
    "c": "=",
    "r": Object {
      "dark_plus": "keyword.operator: #D4D4D4",
      "dark_vs": "keyword.operator: #D4D4D4",
      "hc_black": "keyword.operator: #D4D4D4",
      "hc_light": "keyword.operator: #000000",
      "light_plus": "keyword.operator: #000000",
      "light_vs": "keyword.operator: #000000",
    },
    "t": "source.js meta.var.expr.js keyword.operator.assignment.js",
  },
  Object {
    "c": " ",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js",
  },
  Object {
    "c": "b",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates",
  },
  Object {
    "c": "\`",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates punctuation.definition.string.template.begin.js",
  },
  Object {
    "c": "foo",
    "r": Object {
      "dark_plus": "entity.other.attribute-name: #9CDCFE",
      "dark_vs": "entity.other.attribute-name: #9CDCFE",
      "hc_black": "entity.other.attribute-name: #9CDCFE",
      "hc_light": "entity.other.attribute-name: #264F78",
      "light_plus": "entity.other.attribute-name: #E50000",
      "light_vs": "entity.other.attribute-name: #E50000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.binding entity.other.attribute-name",
  },
  Object {
    "c": ".",
    "r": Object {
      "dark_plus": "keyword.control: #C586C0",
      "dark_vs": "keyword.control: #569CD6",
      "hc_black": "keyword.control: #C586C0",
      "hc_light": "keyword.control: #B5200D",
      "light_plus": "keyword.control: #AF00DB",
      "light_vs": "keyword.control: #0000FF",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.binding keyword.control",
  },
  Object {
    "c": "bar",
    "r": Object {
      "dark_plus": "entity.other.attribute-name: #9CDCFE",
      "dark_vs": "entity.other.attribute-name: #9CDCFE",
      "hc_black": "entity.other.attribute-name: #9CDCFE",
      "hc_light": "entity.other.attribute-name: #264F78",
      "light_plus": "entity.other.attribute-name: #E50000",
      "light_vs": "entity.other.attribute-name: #E50000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.binding entity.other.attribute-name",
  },
  Object {
    "c": "[",
    "r": Object {
      "dark_plus": "meta.embedded: #D4D4D4",
      "dark_vs": "meta.embedded: #D4D4D4",
      "hc_black": "meta.embedded: #FFFFFF",
      "hc_light": "meta.embedded: #292929",
      "light_plus": "meta.embedded: #000000",
      "light_vs": "meta.embedded: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.binding",
  },
  Object {
    "c": "\\"",
    "r": Object {
      "dark_plus": "string: #CE9178",
      "dark_vs": "string: #CE9178",
      "hc_black": "string: #CE9178",
      "hc_light": "string: #0F4A85",
      "light_plus": "string: #A31515",
      "light_vs": "string: #A31515",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.binding string.quoted.double.js punctuation.definition.string.begin.js",
  },
  Object {
    "c": "baz",
    "r": Object {
      "dark_plus": "string: #CE9178",
      "dark_vs": "string: #CE9178",
      "hc_black": "string: #CE9178",
      "hc_light": "string: #0F4A85",
      "light_plus": "string: #A31515",
      "light_vs": "string: #A31515",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.binding string.quoted.double.js",
  },
  Object {
    "c": "\\"",
    "r": Object {
      "dark_plus": "string: #CE9178",
      "dark_vs": "string: #CE9178",
      "hc_black": "string: #CE9178",
      "hc_light": "string: #0F4A85",
      "light_plus": "string: #A31515",
      "light_vs": "string: #A31515",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.binding string.quoted.double.js punctuation.definition.string.end.js",
  },
  Object {
    "c": "]",
    "r": Object {
      "dark_plus": "meta.embedded: #D4D4D4",
      "dark_vs": "meta.embedded: #D4D4D4",
      "hc_black": "meta.embedded: #FFFFFF",
      "hc_light": "meta.embedded: #292929",
      "light_plus": "meta.embedded: #000000",
      "light_vs": "meta.embedded: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.binding",
  },
  Object {
    "c": "\`",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates punctuation.definition.string.template.end.js",
  },
  Object {
    "c": ";",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js punctuation.terminator.statement.js",
  },
]
`;

exports[`Syntax Highlight Syntax Highlight: basic-expression.js 1`] = `
Array [
  Object {
    "c": "const",
    "r": Object {
      "dark_plus": "storage.type: #569CD6",
      "dark_vs": "storage.type: #569CD6",
      "hc_black": "storage.type: #569CD6",
      "hc_light": "storage.type: #0F4A85",
      "light_plus": "storage.type: #0000FF",
      "light_vs": "storage.type: #0000FF",
    },
    "t": "source.js meta.var.expr.js storage.type.js",
  },
  Object {
    "c": " ",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js",
  },
  Object {
    "c": "test1",
    "r": Object {
      "dark_plus": "variable.other.constant: #4FC1FF",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "variable: #9CDCFE",
      "hc_light": "variable.other.constant: #02715D",
      "light_plus": "variable.other.constant: #0070C1",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js meta.var-single-variable.expr.js meta.definition.variable.js variable.other.constant.js",
  },
  Object {
    "c": " ",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js meta.var-single-variable.expr.js",
  },
  Object {
    "c": "=",
    "r": Object {
      "dark_plus": "keyword.operator: #D4D4D4",
      "dark_vs": "keyword.operator: #D4D4D4",
      "hc_black": "keyword.operator: #D4D4D4",
      "hc_light": "keyword.operator: #000000",
      "light_plus": "keyword.operator: #000000",
      "light_vs": "keyword.operator: #000000",
    },
    "t": "source.js meta.var.expr.js keyword.operator.assignment.js",
  },
  Object {
    "c": " ",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js",
  },
  Object {
    "c": "e",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates",
  },
  Object {
    "c": "\`",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates punctuation.definition.string.template.begin.js",
  },
  Object {
    "c": "foo",
    "r": Object {
      "dark_plus": "entity.name.function: #DCDCAA",
      "dark_vs": "meta.embedded: #D4D4D4",
      "hc_black": "entity.name.function: #DCDCAA",
      "hc_light": "entity.name.function: #5E2CBC",
      "light_plus": "entity.name.function: #795E26",
      "light_vs": "meta.embedded: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.expression entity.name.function.js",
  },
  Object {
    "c": "()",
    "r": Object {
      "dark_plus": "keyword.operator: #D4D4D4",
      "dark_vs": "keyword.operator: #D4D4D4",
      "hc_black": "keyword.operator: #D4D4D4",
      "hc_light": "keyword.operator: #000000",
      "light_plus": "keyword.operator: #000000",
      "light_vs": "keyword.operator: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.expression keyword.operator.eq",
  },
  Object {
    "c": " ",
    "r": Object {
      "dark_plus": "meta.embedded: #D4D4D4",
      "dark_vs": "meta.embedded: #D4D4D4",
      "hc_black": "meta.embedded: #FFFFFF",
      "hc_light": "meta.embedded: #292929",
      "light_plus": "meta.embedded: #000000",
      "light_vs": "meta.embedded: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.expression",
  },
  Object {
    "c": "+",
    "r": Object {
      "dark_plus": "keyword.operator: #D4D4D4",
      "dark_vs": "keyword.operator: #D4D4D4",
      "hc_black": "keyword.operator: #D4D4D4",
      "hc_light": "keyword.operator: #000000",
      "light_plus": "keyword.operator: #000000",
      "light_vs": "keyword.operator: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.expression keyword.operator.arithmetic.js",
  },
  Object {
    "c": " ",
    "r": Object {
      "dark_plus": "meta.embedded: #D4D4D4",
      "dark_vs": "meta.embedded: #D4D4D4",
      "hc_black": "meta.embedded: #FFFFFF",
      "hc_light": "meta.embedded: #292929",
      "light_plus": "meta.embedded: #000000",
      "light_vs": "meta.embedded: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.expression",
  },
  Object {
    "c": "bar",
    "r": Object {
      "dark_plus": "entity.name.function: #DCDCAA",
      "dark_vs": "meta.embedded: #D4D4D4",
      "hc_black": "entity.name.function: #DCDCAA",
      "hc_light": "entity.name.function: #5E2CBC",
      "light_plus": "entity.name.function: #795E26",
      "light_vs": "meta.embedded: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.expression entity.name.function.js",
  },
  Object {
    "c": "()",
    "r": Object {
      "dark_plus": "keyword.operator: #D4D4D4",
      "dark_vs": "keyword.operator: #D4D4D4",
      "hc_black": "keyword.operator: #D4D4D4",
      "hc_light": "keyword.operator: #000000",
      "light_plus": "keyword.operator: #000000",
      "light_vs": "keyword.operator: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates meta.embedded.block.expression keyword.operator.eq",
  },
  Object {
    "c": "\`",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js meta.var.expr.js taggedTemplates punctuation.definition.string.template.end.js",
  },
  Object {
    "c": ";",
    "r": Object {
      "dark_plus": "default: #D4D4D4",
      "dark_vs": "default: #D4D4D4",
      "hc_black": "default: #FFFFFF",
      "hc_light": "default: #292929",
      "light_plus": "default: #000000",
      "light_vs": "default: #000000",
    },
    "t": "source.js punctuation.terminator.statement.js",
  },
]
`;
